CMakeLists.txt

```diff
@@ -49,8 +49,20 @@ endif()
 if(NOT CMAKE_VERSION VERSION_LESS 3.4.0 AND NOT CMAKE_SYSTEM_NAME STREQUAL "Android")
     set(LANG CXX)
 endif()
+
+
+option(HUNTER_ENABLED "Get third parties (Corrade, glfw, SDL2) via hunter " OFF)
+if (HUNTER_ENABLED)
+    set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_LIST_DIR}/package/hunter")
+    include(HunterInit)
+endif()
+
 project(Magnum ${LANG})
 
+if(HUNTER_ENABLED)
+    include(HunterFetchDeps)
```

Comment:
It was a request of the author to have the calls to hunter_add_package in  a separate location, so that the packaging info is grouped (magnum handles packaging for conan, homebrew, debian, gentoo, and archlinux and they are nicely grouped inside magnum/package).

My tests showed that it has worked on all platforms : some of them have `WITH_AUDIO`, some do not; same for `WITH_GLFWAPPLICATION` and it works on all combinations.

Reply:
@ruslo I never experienced any issues related to the `WITH_*` options being used before the actual `option()`. The options are either present on the command-line `-DWITH_*`, in which case they're already available, or they are present in CMake cache from previous runs. Again, as @pthom says, this was my request to have it in a separate location (previously it was done like you suggest).

Reply:
> The options are either present on the command-line -DWITH_*, in which case they're already available, or they are present in CMake cache from previous runs.

Okay, I see, that's because all of them set to `OFF` by default. It will not work if you change one of them to `ON`.

Reply:
Yes, true. All features that have external dependencies (and thus need to be explicitly handled by Hunter) are `OFF` by default as a rule.